### ISSUE TITLE
[MPS] MPSGraph normalization is significantly slower for rank-5 tensors

### ISSUE THREAD

**Issue by nuri-yoo** [pytorch/pytorch#180334]
MPSGraph's normalization ops (`normalizationWithTensor:`, `normalizationGradientWithIncomingGradientTensor:`) are significantly slower for rank-5 (BatchNorm3d) inputs compared to rank-4 (BatchNorm2d) with the same number of elements.

Since BatchNorm reduces over all dimensions except the channel dim, spatial dimensions can be safely merged: `[N, C, D, H, W]` → `[N, C, D*H, W]`.

### Benchmark (M4 Pro, PyTorch main @ 0b9170d)

```
native_batch_norm forward (shape: [4, 64, 64, 64, 64]):
  5D [N,C,D,H,W]:     4.5 ms
  4D [N,C,D*H,W]:     4.5 ms  ← after flatten

native_batch_norm_backward:
  5D [N,C,D,H,W]:     6.7 ms
  4D [N,C,D*H,W]:     6.8 ms  ← after flatten

nn.BatchNorm3d fwd+bwd (before fix):  8.7 ms
nn.BatchNorm2d fwd+bwd (equivalent):  3.5 ms  ← 2.4x faster
```

### Reproducer

```python
import time
import torch
import torch.nn as nn

assert torch.backends.mps.is_available()
device = torch.device("mps")

N, C, D, H, W = 4, 64, 64, 64, 64
steps = 20

def bench(label, fn):
    for _ in range(5):
        fn()
    torch.mps.synchronize()
    t0 = time.perf_counter()
    for _ in range(steps):
        fn()
    torch.mps.synchronize()
    ms = (time.perf_counter() - t0) / steps * 1000
    print(f"{label}: {ms:.1f} ms")
    return ms

bn3d = nn.BatchNorm3d(C).to(device).train()
bn2d = nn.BatchNorm2d(C).to(device).train()
bn2d.load_state_dict(bn3d.state_dict())

x5 = torch.randn(N, C, D, H, W, device=device, requires_grad=True)
x4 = torch.randn(N, C, D*H, W, device=device, requires_grad=True)

t5 = bench("BatchNorm3d (5D) fwd+bwd", lambda: bn3d(x5).sum().backward())
t4 = bench("BatchNorm2d (4D) fwd+bwd", lambda: bn2d(x4).sum().backward())
print(f"Ratio: {t5/t4:.1f}x")
```

cc @jerryzh168 @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk
